app/menus/EmptyTrashMenu.tsx

```diff
@@ -0,0 +1,111 @@
+import { observer } from "mobx-react";
+import { TrashIcon } from "outline-icons";
+import * as React from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { useHistory } from "react-router";
+import Document from "~/models/Document";
+import Button from "~/components/Button";
+import Flex from "~/components/Flex";
+import Modal from "~/components/Modal";
+import Text from "~/components/Text";
+import useStores from "~/hooks/useStores";
+import useToasts from "~/hooks/useToasts";
+
+function EmptyTrashMenu() {
+  const { t } = useTranslation();
+  const { documents } = useStores();
+  const [showModal, setShowModal] = React.useState(false);
+  const [isDeleting, setIsDeleting] = React.useState(false);
+  const { showToast } = useToasts();
+  const history = useHistory();
+
+  const [trashed, setTrashed] = React.useState<Document[]>([]);
+
+  React.useEffect(() => {
+    async function getTrashed() {
+      const trashedDocs = await documents.fetchDeleted();
```

Comment:
I think in hindsight the ticket should have been more detailed, while this will work on smaller amounts of docs it has a couple of problems:

- Many serial round-trips to the server
- The documents will gradually disappear from view as the iteration happen
- If there are more than a single page of deleted documents then it will not actually empty the trash

This can all be resolved by changing the implementation so there is an API endpoint to delete all of the documents in one go

Reply:
Very valid points. 

It looks like most of the infrastructure requirements are already in place for this feat. `documentPermanentDeleter` also takes in `Document[]`.

So in brief - 
1. add a new route to `routes/api/documents`, 
2. make sure `user` is `admin`, 
3. find all trashed documents (with `paranoid: false`)
4. call `documentPermanentDeleter`
5. emit event(s) (single `documents.trash_empty` or multiple `documents.permanent_delete`?)

> API endpoint to delete all of the documents

What should we call the endpoint? `documents.empty_trash` maybe?

Reply:
Yes, use `findAllInBatches` and call `documentPermanentDeleter` as many times as needed – always need to protect against very large queries, someone will put 30,000 issues in trash.

> What should we call the endpoint? 

`documents.empty_trash` works for me

Reply:
Let me know if I am on the right track with these latest commits!

Reply:
Sometimes setting PR to draft suppresses notification so @tommoor just in case. Could use your opinion on the data transfer from [here](https://github.com/outline/outline/blob/75d1329c2e548a897c130476a65050f9742e5b83/server/routes/api/documents.ts#L1121) to [here](https://github.com/outline/outline/blob/75d1329c2e548a897c130476a65050f9742e5b83/app/stores/DocumentsStore.ts#L691). I was pondering splitting it into two endpoints - one would fetch info about imminently deleted docs and other would perform the action.